mimic/test/behavior_tests.py

```diff
@@ -0,0 +1,213 @@
+    2 will ignored.  Example::
+
+        # tuples of (name of the behavior, parameters for the behavior)
+        [
+            ("fail", {"code": 500, "message": "Stuff is broken"}),
+            ("fail", {"code": 404, "message": "Stuff doesn't exist"})
+        }
+    """)
+
+    def trigger_event():
+        """
+        Make a request to the regular API that this behavior event modifies.
+        Note that the event should conform to the criteria given.
+
+        :return: a tuple of (response, body)
+        """
+
+    def validate_injected_behavior(name_and_params, response, body):
+        """
+        Validate that the named behavior was triggered as opposed to the
+        default behavior or some other behavior.
+
+        :param name_and_params: The name and parameters of the behavior to
+            trigger - this will be one of the ones specified in
+            :ivar:`name_and_params`.
+        :param response: The response from triggering the event.
+        :param body: The response body from triggering the event.
+        """
+
+    def validate_default_behavior(response, body):
+        """
+        Validate that the default behavior was triggered as opposed to
+        some other behavior.
+
+        :param response: The response from triggering the event.
+        :param body: The response body from triggering the event.
+        """
+
+    def from_test_case(test_case):
+        """
+        A constructor that generates a provider of
+        :class:`IBehaviorAPITestHelper`.
+
+        :param test_case: An instance of a test case that has assertion
+            and cleanup functions.
+        :return: a :class:`IBehaviorAPITestHelper` provider
+        """
+
+
+def register_behavior(test_case, root, uri, behavior_name, parameters,
+                      criteria):
+    """
+    Register a particular behavior.
+
+    :param test_case: the test case with which to make assertions
+    :param root: A mimic root API object
+    :param str uri: The uri fo the behavior resource to register.
+    :param str behavior_name: The name of the behavior
+    :param dict parameters: A dictionary of parameters to pass to the behavior
+    :param list criteria: The criteria for which this behavior should be
+        applied.
+
+    :return: The behavior ID of the registered behavior.
+    """
+    behavior_json = {"name": behavior_name,
+                     "parameters": parameters,
+                     "criteria": criteria}
+    response, body = test_case.successResultOf(json_request(
+        test_case, root, "POST", uri, json.dumps(behavior_json)))
+
+    test_case.assertEqual(response.code, 201)
+    behavior_id = body.get("id")
+    test_case.assertIsInstance(behavior_id, string_types)
+    test_case.assertEqual(UUID(behavior_id).version, 4)
+    return behavior_id
+
+
+def make_behavior_tests(behavior_helper_klass):
+    """
+    Will generate a test suite containing:
```

Comment:
Two minor nitpicks here:

1. beginning with a verb like "will" or "should" or "does" or "handles" (etc) doesn't add information.  "Generate a test suite containing:" is equally helpful.
2. every bullet point starts with "1 test case" and then either implicitly or explicitly "validating".  A cleaner parallel construction would get rid of that and just say "Generate a test suite containing test cases that validate:"